### DISCUSSION TITLE
What's the benefit of using a channel as an input in v2?

### DISCUSSION THREAD

**Discussion by dschow** [cinar/indicator#198] [Q&A] [answered]
I'm messing around with this library and I'm not sure if I'm fully understanding the benefit of passing in a channel. Take this example:

```
inputs := make(chan float64, 15)
inputs <- 1.0
inputs <- -1.0
rsi := momentum.NewRsi[float64]()
result := rsi.Compute(channels[0])
value := <-result
fmt.Println(value)
```

This would make a lot of sense to me if the RSI struct maintained a state and the input channel only represented NEW values. However, it looks like RSI is stateless so every time I call `rsi.Compute` have to pass in a new channel with 15 items which `Compute` consumes from the channel. If I want to compute the RSI on some interval (e.g. once per second) I need to recreate the last N inputs from some other source of truth. What's the benefit here if RSI doesn't hold state and it's consuming the channel at the same time? I feel like I'm missing the benefit of using a channel over a slice.

Originally, I was hoping that RSI did hold state and I could have one goroutine adding new inputs and another goroutine calling `Compute` on some interval, which makes sense to me because in that use-case the channel is how the two goroutines communicate.

**Comment by cinar** [comment, marked as answer]
The Compute method maintains the state, so as you provide more input values through the input channel, you'll get more output values. So, don't call Compute multiple times, just continue using the input and result channels in your code.

The input channel can get values from a file, or from the network, and as new input values are available, RSI will compute the next values and return them to you through the result channel.

**Reply by dschow** [reply]
I think the reason it wasn't working before was because I was only feeding in positive changes so the (average gain/average loss) returned Nan
